cpp/tests/reductions/reduction_tests.cpp

```diff
@@ -96,60 +93,80 @@ struct ReductionTest : public cudf::test::BaseFixture {
   ~ReductionTest() {}
 
   template <typename T_out>
-  void reduction_test(const cudf::column_view underlying_column,
-                      T_out expected_value,
-                      bool succeeded_condition,
-                      std::unique_ptr<reduce_aggregation> const& agg,
-                      cudf::data_type output_dtype = cudf::data_type{},
-                      bool expected_null           = false)
+  std::pair<T_out, bool> reduction_test(cudf::column_view const& underlying_column,
+                                        std::unique_ptr<reduce_aggregation> const& agg,
+                                        cudf::data_type output_dtype = cudf::data_type{})
   {
     if (cudf::data_type{} == output_dtype) output_dtype = underlying_column.type();
+    std::unique_ptr<cudf::scalar> reduction = cudf::reduce(underlying_column, agg, output_dtype);
+    using ScalarType                        = cudf::scalar_type_t<T_out>;
+    auto result                             = static_cast<ScalarType*>(reduction.get());
+    return {result->value(), result->is_valid()};
+  }
 
-    auto statement = [&]() {
-      std::unique_ptr<cudf::scalar> result = cudf::reduce(underlying_column, agg, output_dtype);
-      using ScalarType                     = cudf::scalar_type_t<T_out>;
-      auto result1                         = static_cast<ScalarType*>(result.get());
-      EXPECT_EQ(expected_null, !result1->is_valid());
-      if (result1->is_valid()) { EXPECT_EQ(expected_value, T_out{result1->value()}); }
-    };
-
-    if (succeeded_condition) {
-      CUDF_EXPECT_NO_THROW(statement());
-    } else {
-      EXPECT_ANY_THROW(statement());
-    }
+  // Test with initial value
+  template <typename T_out>
+  std::pair<T_out, bool> reduction_test(cudf::column_view const& underlying_column,
+                                        cudf::scalar const& initial_value,
+                                        std::unique_ptr<reduce_aggregation> const& agg,
+                                        cudf::data_type output_dtype = cudf::data_type{})
+  {
+    if (cudf::data_type{} == output_dtype) output_dtype = underlying_column.type();
```

Comment:
```suggestion
                                        std::optional<cudf::data_type> _output_dtype = {})
  {
    auto const output_dtype = _output_dtype.value_or(underlying_column.type());
```
Not happy with variable names, but something like this seems cleaner. I certainly will not hold up this PR over it.